lib/charms/opensearch/v0/opensearch_relation_peer_cluster.py

```diff
@@ -858,6 +862,11 @@ def _error_set_from_tls(self, peer_cluster_rel_data: PeerClusterRelData) -> bool
                 blocked_msg = "CA certificate mismatch between clusters."
                 should_sever_relation = True
 
+        if not peer_cluster_rel_data.credentials.admin_tls["truststore-password"]:
+            logger.info("Relation data for TLS is missing.")
+            blocked_msg = "CA truststore-password not available."
+            should_sever_relation = True
+
```

Comment:
In large deployments it can happen that a non-main-orchestrator is up before the main-orchestrator provides the admin-tls data. If this is the case, the non-main-orchestrator should be blocked until the main-orchestrator provides the data. If this doesn't happen, it could be that the non-main-orchestrator creates an own truststore password that will later be overwritten.